src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java

```diff
@@ -4370,6 +4370,65 @@ public void clearFindResults(ServerRequestCallback<Void> requestCallback)
       sendRequest(RPC_SCOPE, "clear_find_results", requestCallback);
    }
    
+   @Override
+   public void previewReplace(String searchString,
+                              boolean searchRegex,
+                              boolean searchIgnoreCase,
+                              FileSystemItem directory,
+                              JsArrayString filePatterns,
+                              String replaceString,
+                              boolean replaceRegex,
+                              boolean gitIgnore,
+                              ServerRequestCallback<String> requestCallback)
+   {
+      JSONArray params = new JSONArray();
+      params.set(0, new JSONString(searchString));
+      params.set(1, JSONBoolean.getInstance(searchRegex));
+      params.set(2, JSONBoolean.getInstance(searchIgnoreCase));
+      params.set(3, new JSONString(directory == null ? ""
+                                                     : directory.getPath()));
+      params.set(4, new JSONArray(filePatterns));
+      params.set(5, new JSONString(replaceString));
+      params.set(6, JSONBoolean.getInstance(replaceRegex));
+      params.set(7, JSONBoolean.getInstance(gitIgnore));
+
+      sendRequest(RPC_SCOPE, "preview_replace", params, requestCallback);
```

Comment:
Interesting that BEGIN_FIND and STOP_FIND are set up this way but not clear_find_results. I've updated the code to use a constant variable since it seems to be the most common style.